systemtests/pom.xml

```
@@ -0,0 +1,226 @@
+                    <execution>
+                        <id>system-tests</id>
+                        <goals>
+                            <goal>verify</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <skipTests>${skipSTs}</skipTests>
+                    <includes>
+                        <include>**/*ST.java</include>
+                    </includes>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+                <executions>
+                    <!-- Attach JaCoCo agent to unit tests -->
+                    <execution>
+                        <id>default-prepare-agent</id>
+                        <goals>
+                            <goal>prepare-agent</goal>
+                        </goals>
+                        <configuration>
+                            <destFile>${project.build.directory}/jacoco-unit.exec</destFile>
+                            <append>true</append>
+                        </configuration>
+                    </execution>
+                    <!-- Generate aggregated coverage report -->
+                    <execution>
+                        <id>report-aggregate</id>
+                        <goals>
+                            <goal>report-aggregate</goal>
+                        </goals>
+                        <phase>verify</phase>
+                        <configuration>
+                            <excludes>
+                                <exclude>**/*Builder.class</exclude>
+                                <exclude>**/*Fluent.class</exclude>
+                                <exclude>**/*Nested.class</exclude>
+                                <!-- Exclude Constants and Enums -->
+                                <exclude>com/github/streamshub/systemtests/constants/*</exclude>
+                                <exclude>com/github/streamshub/systemtests/enums/*</exclude>
+                                <!-- Exclude Environment. It's tested by TF -->
+                                <exclude>com/github/streamshub/systemtests/Environment.class</exclude>
+                                <!-- Exclude LogCollector and Logger from code coverage -->
+                                <exclude>com/github/streamshub/systemtests/logs/*</exclude>
+                                <!-- Exclude unit test utils -->
+                                <exclude>com/github/streamshub/systemtests/unit/*</exclude>
+                            </excludes>
+                            <includeCurrentProject>true</includeCurrentProject>
+                            <outputDirectory>${project.reporting.outputDirectory}/jacoco</outputDirectory>
+                            <formats>
+                                <format>HTML</format>
+                            </formats>
```

Sonar needs the XML report format.
```suggestion
                            <formats>
                                <format>HTML</format>
                                <format>XML</format>
                            </formats>
```